src/utils/calculations.spec.ts

```diff
@@ -0,0 +1,42 @@
+import { PartialDeep } from 'type-fest';
```

Comment:
For the sake of not adding another package, do you think Partial<T> can be enough here? Just trying to find a balance between the value it can add to have more type safety vs adding a dependency. Tree shaking should take care of dependency in the build, so it should not increase the size of the bundled js.

Reply:
`Partial` is not enough here, we can just define the `PartialDeep` ourselves, but that `type-fest` is commonly used.

Reply:
I would say that if we only use this one, we can just define the type ourselves yes. If later we see we need more special types, then we can think of adding the whole lib...

@turtlemoji what do you think?

Reply:
Using our own `DeepPartial` with https://github.com/yearn/yearn-finance-v3/pull/562/commits/59f6de9dabffff9a8f7f80b5daf3daa0209908fc, `type-fest` removed 😢

Reply:
Having the type ourselves and not having to think about prototype pollution or dependency injection in another lib is amazing for me.
If it was for me just even trim lodash too 😈  @xgambitox @karelianpie

Reply:
+1 for no lodash for me as well :)

Reply:
I <3 lodash